race-buffer/src/writer.rs

```diff
@@ -0,0 +1,252 @@
+use crate::{get_cursor_index, Entry};
+use core::iter::Iterator;
+use core::mem::MaybeUninit;
+
+#[cfg(not(feature = "std"))]
+use core::fmt;
+#[cfg(feature = "std")]
```

Comment:
It's fine to just use `core::fmt` even when the standard library is available.  Can remove some configuration flagging in this vein.